pkg/activator/net/revision_backends.go

```diff
@@ -452,6 +452,14 @@ func (rbm *revisionBackendsManager) getOrCreateRevisionWatcher(rev types.Namespa
 	rbm.revisionWatchersMux.Lock()
 	defer rbm.revisionWatchersMux.Unlock()
 
+	// Don't create a watcher if we're already shutting down. Crucial to be called
+	// called under lock to be consistent with the above.
+	select {
+	case <-rbm.ctx.Done():
```

Comment:
I wonder if returning `errShutdown` or something is more consistent (and would not require an err check below)

Reply:
I think that'd run into similar races because of the logging involved :thinking:

Reply:
May be we can tweak the test? :)